sql/core/src/test/scala/org/apache/spark/sql/execution/benchmark/NestedSchemaPruningBenchmark.scala

```diff
@@ -33,13 +33,17 @@ abstract class NestedSchemaPruningBenchmark extends SqlBasedBenchmark {
   protected val N = 1000000
   protected val numIters = 10
 
-  // We use `col1 BIGINT, col2 STRUCT<_1: BIGINT, _2: STRING>` as a test schema.
-  // col1 and col2._1 is used for comparision. col2._2 mimics the burden for the other columns
+  // We use `col1 BIGINT, col2 STRUCT<_1: BIGINT, _2: STRING>,
+  // col3 ARRAY<STRUCT<_1: BIGINT, _2: STRING>>` as a test schema.
+  // col1, col2._1 and col3._1 are used for comparision. col2._2 and col3._2 mimics the burden
+  // for the other columns
   private val df = spark
     .range(N * 10)
     .sample(false, 0.1)
-    .map(x => (x, (x, s"$x" * 100)))
-    .toDF("col1", "col2")
+    .map { x =>
+      val col3 = (0 until 10).map(i => (x + i, s"$x" * 10))
+      (x, (x, s"$x" * 100), col3)
+    }.toDF("col1", "col2", "col3")
```

Comment:
Ur, this seems to have a side-effect on the `Nested column`, too. I ran the test at AS-IS PR. The following is the result.

**PREVIOUS**
```
OpenJDK 64-Bit Server VM 1.8.0_201-b09 on Linux 3.10.0-862.3.2.el7.x86_64
Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
------------------------------------------------------------------------------------------------------------------------
Top-level column                                    131            150          25          7.7         130.6       1.0X
Nested column                                       922            954          21          1.1         922.2       0.1X
```

**THIS PR**
```
[info] OpenJDK 64-Bit Server VM 1.8.0_212-b04 on Linux 3.10.0-862.3.2.el7.x86_64
[info] Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
[info] Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
[info] ------------------------------------------------------------------------------------------------------------------------
[info] Top-level column                                    137            160          23          7.3         136.5       1.0X
[info] Nested column                                      1211           1243          27          0.8        1210.5       0.1X
[info] Nested column in array                            11210          11292          39          0.1       11209.6       0.0X
```

Reply:
Also, could you rebase to the master?

Reply:
Thanks @dongjoon-hyun. I made the array less heavy. 

Btw, previously the test creates temp views `t1`, `t2` for benchmark cases, individually. The added case reuses `t2`, but to follow with previous cases, I created `t3` now.

Also compared the query plans from previous and this PR. They are actually the same locally.

Previous:
```
[info] == Physical Plan ==                                                                                                                             
[info] *(1) Project [col2#26._1 AS _1#239L]                                                                                                            
[info] +- *(1) FileScan orc [col2#26] Batched: false, DataFilters: [], Format: ORC, Location: InMemoryFileIndex[file:/spark/target/tmp/spark-3e4433f7-c92e-4017-9efb-3bbd2..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<col2:struct<_1:bigint>>              
```

This PR:
```
[info] == Physical Plan ==                                                                                                                             
[info] *(1) Project [col2#26._1 AS _1#231L]                                                                                                            
[info] +- *(1) FileScan orc [col2#26] Batched: false, DataFilters: [], Format: ORC, Location: InMemoryFileIndex[file:/spark/target/tmp/spark-11a5c59b-5b37-4eff-8fdd-398c7..., PartitionFilters: [], PushedFilters: [], ReadSchema: struct<col2:struct<_1:bigint>>                       
```

The benchmark results don't differ in local test.

Previous:
```
[info] Java HotSpot(TM) 64-Bit Server VM 1.8.0_202-b08 on Mac OS X 10.14.5                                                                             
[info] Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz                                                                                                        
[info] Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative                        
[info] ------------------------------------------------------------------------------------------------------------------------                        
[info] Top-level column                                     86            120          48         11.6          86.4       1.0X                        
[info] Nested column                                      1081           1097           9          0.9        1080.9       0.1X        
```

This PR:
```
[info] Java HotSpot(TM) 64-Bit Server VM 1.8.0_202-b08 on Mac OS X 10.14.5                                                                             
[info] Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz                                                                                                        
[info] Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative                        
[info] ------------------------------------------------------------------------------------------------------------------------                        
[info] Top-level column                                     80             89          10         12.5          79.9       1.0X                        
[info] Nested column                                      1038           1052           9          1.0        1038.4       0.1X                        
[info] Nested column in array                             2576           2603          18          0.4        2575.7       0.0X              
```

Reply:
Thanks. Yes. Let me check that tomorrow again.

Reply:
The regression is on master branch. For Parquet, it looks good. But ORC, I got the following. 
```
 Nested Schema Pruning Benchmark For ORC v1
 ================================================================================================

-OpenJDK 64-Bit Server VM 1.8.0_201-b09 on Linux 3.10.0-862.3.2.el7.x86_64
+OpenJDK 64-Bit Server VM 1.8.0_212-b04 on Linux 3.10.0-862.3.2.el7.x86_64
 Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
 Selection:                                Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Top-level column                                    131            150          25          7.7         130.6       1.0X
-Nested column                                       922            954          21          1.1         922.2       0.1X
+Top-level column                                    125            162          23          8.0         125.5       1.0X
+Nested column                                      1210           1272          92          0.8        1209.9       0.1X

-OpenJDK 64-Bit Server VM 1.8.0_201-b09 on Linux 3.10.0-862.3.2.el7.x86_64
+OpenJDK 64-Bit Server VM 1.8.0_212-b04 on Linux 3.10.0-862.3.2.el7.x86_64
 Intel(R) Xeon(R) CPU E5-2670 v2 @ 2.50GHz
 Limiting:                                 Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
 ------------------------------------------------------------------------------------------------------------------------
-Top-level column                                    446            477          50          2.2         445.5       1.0X
-Nested column                                      1328           1366          44          0.8        1328.4       0.3X
+Top-level column                                    432            468          30          2.3         432.1       1.0X
+Nested column                                      1632           1685          77          0.6        1631.5       0.3X
```

Reply:
Please ignore the above.